### PR TITLE
chore: convert @fluid-experimental/property-common to TS [ 2 /2 ]

### PR DESCRIPTION
**Pull request by evaliyev**

### CODE REVIEW

**Comment by evaliyev** [comment]
@karlbom @nedalhy PTAL
@DLehenbauer FYI

File: experimental/PropertyDDS/packages/property-common/src/datastructures/integer64.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ * integers natively yet.
+ *
+ * Please note this class is immutable.
+ * There are and there won't be set functions!
+ * (see javascript String class)
+ *
+ * @param low  - lower 32 bit
+ * @param high - higher 32 bit
+ */
+export class Integer64 {
```
Comment (DLehenbauer) [inline experimental/PropertyDDS/packages/property-common/src/datastructures/integer64.ts:28]: FYI - I haven't had a chance to try them, but we now have BigInt and BigInt/Uint64Array in JavaScript.

Comment (nedalhy) [reply experimental/PropertyDDS/packages/property-common/src/datastructures/integer64.ts:28]: @ruiterr FYI

File: experimental/PropertyDDS/packages/property-common/src/events.ts [hunk 2]
```diff
@@ -1,543 +0,0 @@
-/*!
```
Comment (ruiterr) [inline experimental/PropertyDDS/packages/property-common/src/events.ts]: Why has this file been removed? It is not needed anywhere?

File: experimental/PropertyDDS/packages/property-common/src/test/datastructures/collection.spec.ts [hunk 3]
```diff
@@ -439,65 +411,50 @@ describe('collection', function () {
...
         collection2.copy(collection);
 
-        expect(collection2.items['item1']).to.equal(objectToAdd['item1']);
-        expect(collection2.items['item2']).to.equal(objectToAdd['item2']);
-        expect(collection2.items['item3']).to.equal(objectToAdd['item3']);
-
-        done();
-    });
-
-    it('should respect the type of Collection given', function (done) {
```
Comment (ruiterr) [inline experimental/PropertyDDS/packages/property-common/src/test/datastructures/collection.spec.ts]: I assume this test has been removed, because the type check had been removed earlier. Why did you remove the type check?

**Review by ruiterr — APPROVED** [review]
I posted several questions, please have a look at those. I assume that the removed files / functions were not needed anywhere else? Apart from that, I think these changes are ok.

File: experimental/PropertyDDS/packages/property-common/src/events.ts (hunk 2 above)
Comment (evaliyev) [reply experimental/PropertyDDS/packages/property-common/src/events.ts]: Using built-in "events" package from TS.

File: experimental/PropertyDDS/packages/property-common/src/error_objects/operationError.ts [hunk 4]
```diff
@@ -2,106 +2,89 @@
...
+       * @param flags - Flags that characterize the error. See {@link FlaggedError.FLAGS}.
+       */
+    constructor(
+        message?: string,
+        public operation?: string,
+        public statusCode?: number,
+        public flags: number = 0,
+    ) {
+        super(message);
+        Object.setPrototypeOf(this, OperationError.prototype);
```
[the remark this answers is not in the document]
Comment (evaliyev) [reply experimental/PropertyDDS/packages/property-common/src/error_objects/operationError.ts:35]: That is the working inheritance from the Error class.
https://stackoverflow.com/questions/41102060/typescript-extending-error-class
Failing existing tests otherwise.

File: experimental/PropertyDDS/packages/property-common/src/constants.ts [hunk 5]
```diff
@@ -96,544 +21,395 @@ const VALIDATOR_MSG = {
         'SV-001: Only properties that inherit from NamedProperty can have a context of "set". typeid: ',
```
[the remark this answers is not in the document]
Comment (evaliyev) [reply experimental/PropertyDDS/packages/property-common/src/constants.ts:21]: yes!

File: experimental/PropertyDDS/packages/property-common/src/datastructures/collection.ts [hunk 6]
```diff
@@ -5,512 +5,416 @@
...
-    if (this._type && !(in_value instanceof this._type)) {
-        throw new Error(MSGS.TYPE_MISMATCH);
-    } else {
-        return true;
+    /**
+     * @param in_key - the key we are looking for
+     * @returns true if the item exists
+     */
+    has(in_key: string | number): boolean {
+        return Object.prototype.hasOwnProperty.call(this._items, in_key);
```
[the remark this answers is not in the document]
Comment (evaliyev) [reply experimental/PropertyDDS/packages/property-common/src/datastructures/collection.ts:253]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty#using_hasownproperty_as_a_property_name

**Comment by nedalhy** [comment]
@DLehenbauer could you please review & merge this PR?

**Review by DLehenbauer — APPROVED** [review]
